pkg/upgrade/v1_18_0.go

```diff
@@ -0,0 +1,75 @@
+package upgrade
+
+import (
+	"context"
+	"fmt"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/jaegertracing/jaeger-operator/pkg/apis/jaegertracing/v1"
+	log "github.com/sirupsen/logrus"
+)
+
+func upgrade1_18_0(ctx context.Context, client client.Client, jaeger v1.Jaeger) (v1.Jaeger, error) {
+	collectorDeprecatedFlags := []deprecationFlagMap{
+		{
+			from: "collector.port",
+			to:   "collector.tchan-server.host-port",
+		},
+		{
+			from: "collector.http-port",
+			to:   "collector.http-server.host-port",
+		},
+		{
+			from: "collector.grpc-port",
+			to:   "collector.grpc-server.host-port",
+		},
+		{
+			from: "collector.zipkin.http-port",
+			to:   "collector.zipkin.host-port",
+		},
+		{
+			from: "admin-http-port",
+			to:   "admin.http.host-port",
+		},
+	}
+
+	agentFlags := []string{
+		"collector.host-port",
+		"reporter.tchannel.discovery.conn-check-timeout",
+		"reporter.tchannel.discovery.min-peers",
+		"reporter.tchannel.host-port",
+		"reporter.tchannel.report-timeout",
+	}
+
+	jaeger.Spec.Collector.Options = migrateDeprecatedOptions(&jaeger, jaeger.Spec.Collector.Options, collectorDeprecatedFlags)
+	jaeger.Spec.Collector.Options = transformCollectorPorts(&jaeger, jaeger.Spec.Collector.Options, collectorDeprecatedFlags)
+
+	// Remove agent flags
+	deleteAgentFlags := []deprecationFlagMap{}
+	for _, item := range agentFlags {
```

Comment:
It's easier to read and probably faster to define the `deleteAgentFlags` directly using `deprecationFlagMap` type.

Reply:
well, In that case I would need to define a lot of ` { from:"deprecated_flag", to:"" }.` I wanted to avoid the repeated `to:""` in the code.

Reply:
iirc: the empty string is a default value it can be omitted.

Reply:
Good point, I did the change.